src/gui/creds/flow2auth.cpp

```diff
@@ -187,6 +190,12 @@ void Flow2Auth::slotPollTimerTimeout()
     QUrlQuery arguments(QString("token=%1").arg(_pollToken));
     requestBody->setData(arguments.query(QUrl::FullyEncoded).toLatin1());
 
+    if (_enforceHttps && QUrl(_pollEndpoint).scheme() != "https") {
```

Comment:
@FlexW Do we really need to go that far? I wonder if we can do the same validation at the point where `pollEndpoint` is received?

In here: `pollEndpoint = json.value("poll").toObject().value("endpoint").toString();`
?

Reply:
Yes, thats also possible.